modules/ROOT/pages/dataweave-cookbook-decode-encode.adoc

```diff
@@ -0,0 +1,160 @@
+//Need to check if PDF input is supported by validator.
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from dw::core::Binaries
+var myPDF = readUrl("classpath://pdf-test.pdf", "application/octet-stream")
+output text/plain
+---
+toBase64(myPDF as Binary)
+----
+
+.Output (partial results):
+[source,json,linenums]
+----
+JVBERi0xLjYNJeLjz9MNCjM3IDAgb2JqIDw8L0xpbmVhcml6ZWQgMS9MIDIwNTk3L08gNDAvRSAx
+...
+----
+
+
+[[ex02]]
+== Decode Base64
+
+This example simulates the transformation in the Write operation from the <<xml_config, XML configuration>> by converting a Base64 file stream (`application/octet-stream`) to Binary and outputting the Binary in the `application/PDF with binary` format. The example does not generate a PDF file; a Write operation is required to generate such a file. 
+
+The script reads the input PDF as an octet-stream MIME type. From within the call to `fromBase64`, the script coerces that input to a string value before coercing the string to Base64.
+
+The examples use the following functions:
+
+* `fromBase64` in the `dw::core::Binaries` module to transform a Base64 string to a binary value.
+* `readUrl` to input a PDF file.
++
+The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `application/octet-stream`. Using the MIME type `"binary"` as the second `readUrl` argument is also valid for this example.
+
+//NOTE that we can't yet automate this example. Outputs text format.
+//Need to check if PDF input is supported by validator.
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from dw::core::Binaries
+var myPDF = readUrl("classpath://pdf-test.pdf", "application/octet-stream")
+var myPDFasBinary = toBase64(myPDF as Binary)
+output application/PDF with binary
+---
+fromBase64(myPDFasBinary as String) as Binary
+----
+
+The output is the text representation of the PDF. 
+
+.Output (partial results):
+[source,json,linenums]
+----
+%PDF-1.6
+%����
+37 0 obj <</Linearized 1/L 20597/O 40/E 14115/N 1/T 19795/H [ 1005 215]>>
+endobj
+                 
+xref
+37 34
+0000000016 00000 n
+...
+----
+
+[[xml_config]]
+== XML Configuration Example
+
+In Anypoint Studio, you can create and run a Mule flow that encodes and decodes a PDF file. After you complete the procedure, the example will read the file from the specified local directory using the Read operation from the File module. It will write the PDF to the specified directory by using the Write operation in the File module. It will use Transform Message components to encode the file to Base64 and to decode that output from Base64.
+
+. Create a Mule project in Anypoint Studio.
+. Use *Add Module* in the Mule Palette to add the File module.
+. Copy and paste the following example between the `<mule/>` elements of a Mule flow in Anypoint Studio.
```

Comment:
```suggestion
To create the Mule flow:

. In Studio, select *File > New > Mule Project*.
. Enter a name for your Mule project and click *Finish*.
. In the *Mule Palette* view, click *(X) Search in Exchange*.
. In the *Add Dependencies to Project* window, type `file` in the search field.
. Click *File Connector* in Available modules.
. Click *Add*.
. Click *Finish*.
. In the Studio canvas, navigate to the *Configuration XML* tab.
. Copy and paste the following XML example between the `<mule/>` elements of the flow:
```

Reply:
done mostly as substeps, keeping high-level steps in place.